src/e2e/java/teammates/e2e/cases/e2e/BaseE2ETestCase.java

```diff
@@ -82,6 +82,12 @@ protected static Url createLocalUrl(String testFileName) throws IOException {
      * Logs in a page using admin credentials (i.e. in masquerade mode).
      */
     protected <T extends AppPage> T loginAdminToPage(AppUrl url, Class<T> typeOfPage) {
+        // When not using dev server, Google blocks log in by automation.
+        // To log in, log in manually to teammates in your browser before running e2e tests.
+        // Refer to teammates.e2e.pageobjects.Browser for more information.
+        if (!TestProperties.isDevServer()) {
+            return null;
```

Comment:
Will this result in NPE?

Reply:
I have changed it to return the AppPage of the url it should navigate to instead.